docs/gadgets/prometheus.md

```diff
@@ -0,0 +1,369 @@
+
+You can see that the counters are already going up for some containers.
+
+Let's create a pod to execute from more processes:
+
+```bash
+$ kubectl run mypod1 -it --image busybox --restart Never -- sh -c 'for i in $(seq 0 1 1000); do cat /dev/null ; ping -c 1 localhost > /dev/null; done'
+```
+
+If we check the counter again, we can see that it shows that our pod has executed a lot of processes:
+
+![Inspektor Gadget Counter Metric](../images/prometheus_counter_2.png)
+
+Now, update the configuration file to only take into considerations executions of the `cat` binary:
+
+```yaml
+metrics:
+  - name: executed_processes
+    type: counter
+    category: trace
+    gadget: exec
+    labels:
+      - namespace
+      - pod
+      - container
+    selector:
+     - "comm:cat"
+```
+
+Restart the gadget
+
+```bash
+$ kubectl gadget prometheus --config @myconfig.yaml --metrics-path /metrics
+INFO[0000] Running. Press Ctrl + C to finish
+INFO[0000] minikube-m02         | Publishing metrics...
+INFO[0000] minikube             | Publishing metrics...
+```
+
+Create a new pod that executes processes:
+
+```bash
+$ kubectl run mypod2 -it --image busybox --restart Never -- sh -c 'for i in $(seq 0 1 1000); do cat /dev/null ; ping -c 1 localhost > /dev/null; done'
+
+```
+
+The counter only takes into consideration the cat commands now:
+
+![Inspektor Gadget Counter Metric](../images/prometheus_counter_3.png)
+
+#### With `ig`
+
+It's also possible to use the prometheus gadget without Kubernetes. In this case, we have to
+configure Prometheus to point to the endpoint exposed by ig, it's `localhost:2223` by default:
+
+```yaml
+# prometheus.yaml
+scrape_configs:
+- job_name: ig
+  scrape_interval: 1s
+  static_configs:
+  - targets:
+    - localhost:2223
+```
+
+Start prometheus with above configuration (please refer to [docker
+installation](https://prometheus.io/docs/prometheus/latest/installation/#using-docker) in case you
+want to run prometheus in a container).
+
+```bash
+$ prometheus --config.file prometheus.yaml
+```
+
+Then, start the prometheus gadget with the same configuration as above Kubernetes section:
+
+```bash
+$ sudo ig prometheus --config @myconfig.yaml
+INFO[0000] Running. Press Ctrl + C to finish
+INFO[0000] Publishing metrics...
+```
+
+<!-- markdown-link-check-disable-next-line -->
+You can check in http://localhost:9090/targets and check that the ig endpoint is reporting metrics:
+
+![ig service up](../images/prometheus_ig_service_up.png)
+
+Let's execute some commands inside a container:
+
+```bash
+docker run --rm -ti --name=mycontainer busybox sh -c 'for i in $(seq 0 1 1000); do cat /dev/null ; ping -c 1 localhost > /dev/null; done'
+```
+
+We can see how the counter for `mycontainer` is increased.
+![ig counter](../images/prometheus_ig_counter1.png)
+
+### Limitations
+
+- The `kubectl gadget` instance has to keep running in order to update the metrics.
+- Histograms aren't supported
```

Comment:
```suggestion
```

See above. IMHO, let's not mention stuff we don't support in the first place.

Reply:
I removed the mention to histograms above, but do you think we should remove this line too?

Reply:
I don't really mind it being there, it just forces us to remember removing it once we add support for histograms.